lib/hooks/queries/useLatestTrades.tsx

```diff
@@ -26,16 +26,26 @@ export type TradeItem = {
   cost: Decimal;
 };
 
-export const useLatestTrades = () => {
+export const useLatestTrades = (limit?: number, marketId?: number) => {
   const [sdk, id] = useSdkv2();
 
   const query = useQuery(
-    [id, transactionHistoryKey],
+    [id, transactionHistoryKey, limit, marketId],
     async () => {
       if (isIndexedSdk(sdk)) {
         const { historicalSwaps } = await sdk.indexer.historicalSwaps({
-          limit: 3,
+          limit: limit,
           order: HistoricalSwapOrderByInput.BlockNumberDesc,
+          ...(marketId != null
+            ? {
+                where: {
```

Comment:
I'm noticing WSX assets creeping in so I think you'll just need to filter out `foreignAsset:3`